apps/web/components/eventtype/EventAdvancedTab.tsx

```diff
@@ -386,20 +392,22 @@ export const EventAdvancedTab = ({ eventType, team }: Pick<EventTypeSetupInfered
                 selectedCustomInput={selectedCustomInput}
                 onSubmit={(values) => {
                   const customInput: CustomInputParsed = {
-                    id: -1,
+                    id: Math.floor(Math.random() * 10000),
                     eventTypeId: -1,
```

Comment:
what is this -1?

Reply:
-1 was used to denote new custom input which was used here to detect whether we have to create a field or update it https://github.com/calcom/cal.com/blob/main/packages/trpc/server/routers/viewer/eventTypes.tsx#L38. i just added another property in the customInputParsed type to denote if we have to create a new input or update it.

Reply:
-1 as id was causing bugs when you create more than one custom input and try to edit them without clicking the save button on top.

Reply:
https://github.com/calcom/cal.com/pull/5959#discussion_r1052525718

Reply:
`-1 * Math.random()` could still give the same negative number for 2 different custom inputs.
The simplest solution that I can think of is to use `Date.now()` as 2 different custom inputs are manually added there would always be a time difference and thus 2 different `Date.now `call will have guaranteed different values.

Reply:
@hariombalhara date.now() was the first thing i tried. there was some prisma error because of the size.

Reply:
we can only send 32 bit signed integer as id just updated the random id generator function to this. This function would give the temporary id till the user clicks save and then we would get the correct id from the server which won't have any collision. 

```
const getRandomId = (length = 8) => {
  return (
    -1 *
    parseInt(
      Math.ceil(Math.random() * Date.now())
        .toPrecision(length)
        .toString()
        .replace(".", "")
    )
  );
};
```